src/main/java/org/noureddine/joularjx/cpu/PowermetricsMacOS.java

```diff
@@ -84,9 +83,22 @@ public double getCurrentPower(double cpuLoad) {
         return 0.0;
     }
 
+    /**
+     * Override point for testing.
+     */
+    protected BufferedReader getReader() {
+        return reader;
+    }
+
     private static int extractPowerInMilliwatts(String line, int powerIndex) {
         try {
-            return Integer.parseInt(line.substring(powerIndex + POWER_INDICATOR_LENGTH, line.indexOf('m') - 1));
+            if (line.trim().endsWith("mW")) {
```

Comment:
I would avoid trimming the line several times, here and use the CPU type detection done previously to avoid checking for these mutually exclusive cases.

Reply:
See above, resolved with splitting into different functions.